test/api/test_module_construct.cpp

```diff
@@ -0,0 +1,39 @@
+#include <migraphx/migraphx.h>
+#include <migraphx/migraphx.hpp>
+#include "test.hpp"
+
+TEST_CASE(add_op)
+{
+    migraphx::program p;
+    migraphx::module m = p.get_main_module();
+    auto x             = m.add_parameter("x", migraphx::shape(migraphx_shape_float_type, {3, 3}));
+    auto y             = m.add_parameter("y", migraphx::shape(migraphx_shape_float_type, {3, 3}));
+    auto add_op        = migraphx::operation("add");
+    m.add_instruction(add_op, {x, y});
+    m.print();
```

Comment:
We should compile and run the program.

Reply:
At least with the `ref` implementation.

Reply:
added logic to run on GPU and verify result with Ref target

Reply:
Modified to run only on Ref.